src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -378,10 +378,12 @@ public static BuildManager DefaultBuildManager
         string IBuildComponentHost.Name => _hostName;
 
         /// <summary>
-        /// Retrieves the build parameters associated with this build.
+        /// Gets or sets the build parameters associated with this build.
         /// </summary>
-        /// <returns>The build parameters.</returns>
-        BuildParameters IBuildComponentHost.BuildParameters => _buildParameters;
+        BuildParameters IBuildComponentHost.BuildParameters
+        {
+            get => _buildParameters; set => _buildParameters = value;
```

Comment:
This change is a little suspicious. I'm not convinced this should be settable

Reply:
Setting them was required when I was testing for missing required properties [here](https://github.com/dotnet/msbuild/pull/8249/commits/5e94b8267897c9547adeb8a9c4cc1ad6d89e8033#diff-b77232ac80795a4839128486d8221283fe6ee537902fbbb0561e8f779e81b7aaR160-R162), which is no longer necessary. Will remove.